src/main/java/com/odysseusinc/athena/service/impl/VocabularyServiceImpl.java

```diff
@@ -206,11 +217,68 @@ public DownloadBundle saveDownloadItems(DownloadBundle bundle, List<Integer> idV
     }
 
     @Override
-    public List<DownloadBundleDTO> getDownloadHistory(Long userId) {
+    public List<DownloadBundleDTO> getDownloadHistory(AthenaUser user) {
 
         Sort sort = new Sort(Sort.Direction.DESC, "created");
-        List<DownloadBundle> history = downloadBundleRepository.findByUserId(userId, sort);
-        return converterUtils.convertList(history, DownloadBundleDTO.class);
+        List<DownloadBundle> history = downloadBundleRepository.findByUserId(user.getId(), sort);
+        Set<Long> bundleIds = history.stream()
+                .map(b -> b.getId())
+                .collect(Collectors.toSet());
+
+        List<DownloadShare> shares = downloadShareRepository.findByDownloadShareIdUserEmail(user.getEmail());
+        List<DownloadBundleDTO> sharedDTOs = new ArrayList<>();
+        // add shared bundles to list of available downloads
+        if (!shares.isEmpty()) {
```

Comment:
How I can share download for the first time?
There is the following condition on the UI side:
```
const isShareable = isAlreadyShared || !bundle.downloadShareDTO;
```
And from server:
```
cdmVersion: 5
date: 1566830435154
downloadShareDTO: null
id: 3
link: null
name: "rxnorm"
releaseVersion: "v5.0 17-JUN-19"
status: "PENDING"
vocabularies: [{id: 8, code: "RxNorm", name: "", required: null, update: "2016-10-03", clickDefault: true,…}]
```